connection.js

```diff
@@ -601,13 +589,169 @@ class Connection {
+    }
+} /* }}} */
+
+function rankToString(r) { /* {{{ */
+    const R = Math.floor(r);
+    if (R >= 30)  return (R-30+1) + 'd'; // R>=30 : 1 dan or stronger
+    else          return (30-R) + 'k';   // R<30  : 1 kyu or weaker
+} /* }}} */
+
+function bannedFamilyReject(argNameString, uid, notificationUid) { /* {{{ */
     const rankedUnranked = beforeRankedUnrankedGamesSpecial("from ", "", argNameString, "all ");
-    conn_log(`Username ${notificationUserUsername}, user id ${notificationUserId}, is banned ${rankedUnranked}`);
-    return { reject: true, msg: `Username ${notificationUserUsername}, user id ${notificationUserId}, is banned ${rankedUnranked} on this bot by bot admin, you may try changing the ranked/unranked setting` };
-}  /* }}} */
+    conn_log(`${uid} ${notificationUid} is banned ${rankedUnranked}`);
+    return { reject: true, msg: `You (${uid} ${notificationUid}) are banned ${rankedUnranked} on this bot by bot admin, you may try changing the ranked/unranked setting` };
+} /* }}} */
+
+function noAutohandicapReject(argNameString) { /* {{{ */
+    const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argNameString, "");
+    conn_log(`no autohandicap ${rankedUnranked}`);
+    return { reject: true, msg: `For easier bot management, -automatic- handicap is disabled on this bot ${rankedUnranked}, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones, you may try changing the ranked/unranked setting` };
+} /* }}} */
+
+function boardsizeNotificationIsNotSquareReject(argNameString, notificationWidth, notificationHeight) { /* {{{ */
+    const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argNameString, "");
+    conn_log(`boardsize ${notificationWidth} x ${notificationHeight} is not square, not allowed ${rankedUnranked}`);
+    return { reject: true, msg: `Your selected board size ${notificationWidth} x ${notificationHeight} is not square, not allowed ${rankedUnranked} on this bot, please choose a SQUARE board size (same width and height), for example try 9x9 or 19x19}` };
+} /* }}} */
+
+function boardsizeSquareToDisplayString(boardsizeSquare) { /* {{{ */
+    return boardsizeSquare
+    .toString()
+    .split(',')
+    .map(e => e.trim())
+    .map(e => `${e}x${e}`)
+    .join(', ');
+} /* }}} */
+
+function pluralFamilyStringToSingularString(plural) { /* {{{ */
+    let pluralToConvert = plural.split("unranked")[0].split("ranked")[0].split("");
+    // for example "speedsranked" -> ["s", "p", "e", "e", "d", "s"]
+    pluralToConvert.pop();
+    // for example ["s", "p", "e", "e", "d", "s"] -> ["s", "p", "e", "e", "d"]
+    pluralToConvert = pluralToConvert.join("");
+    // for example ["s", "p", "e", "e", "d"] -> "speed"
+    return pluralToConvert;
+} /* }}} */
 
-function genericAllowedFamiliesReject(argNameString, notificationUnit) { /* }}} */
+function genericAllowedFamiliesReject(argNameString, notificationUnit) { /* {{{ */
```

Comment:
Ok you fixed the `/* {{{ */`, but tbh, it would be cleaner if we removed all the `/* {{{ */` and `/* }}} */` mess. Also you can reduce the `{{{` cases to `{` (and the corresponding `}}}` to `}`) if you do anything.

Reply:
i have no issue deleting them all, but what do we delete exactly ?
only the reject functions things, or also unrelated functions such as this : https://github.com/wonderingabout/gtp2ogs/blob/devel/connection.js#L576
?